src/System.Buffers/src/System/Buffers/ArrayPoolEventSource.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics.Tracing;
+
+namespace System.Buffers
+{
+    public sealed class ArrayPoolEventSource : EventSource
+    {
+        public ArrayPoolEventSource() : base() { }
+
+        [Event(1, Message = "Rented Buffer {0} of size {1} from Bucket {2}", Level = EventLevel.Informational)]
+        public void BufferRented(int bufferId, int bufferSize, int bucketId) { WriteEvent(1, bufferId, bufferSize, bucketId); }
+
+        [Event(2, Message = "Returned Buffer {0} to Pool", Level = EventLevel.Informational)]
+        public void BufferReturned(int bufferId) { WriteEvent(2, bufferId); }
```

Comment:
I don't know if it makes sense in this case, but EventSource supports a notion of start/stop events that allow tools like PerfView to correlate them automatically and provide information for things like duration.  If we were to logically think of  BufferRented as a start and BufferReturned as a stop, correlated on bufferId, we might be able to more easily get information automatically for things like how long buffers were in use.  I'm not sure it makes sense in this case, but something to investigate.